From bf1978348e411a05ec8e59fa830e5861a8b9ca1b Mon Sep 17 00:00:00 2001
From: Aina Niemetz <aina.niemetz@gmail.com>
Date: Tue, 1 Oct 2024 12:23:35 -0700
Subject: [PATCH 091/312] Bump CaDiCaL version to 2.0.0. (#11231)

Below are the results of version 1.7.4 vs 2.0.0 on non-incremental with
`--sat-solver=cadical`.

We have a performance regression on BV logics with version 2.0.0 (due to
the new version as the BV solver backend). We observe a similar
regression when bumping to 2.0.0 in Bitwuzla. We are working with the
CaDiCaL team to address this regression, but for now we want to take
advantage of new proof features in version 2.0.0.

Interestingly, we have less memouts with version 2.0.0.

(Note: Ignore the errors, they are all parse errors due to a recent
change in the parser wrt variable shadowing, @ajreynol this needs to be
fixed.)
```
                              status   total  solved     sat   unsat    best  timeout  memout  error  uniq  disagr   time_cpu     memory
config
2024-09-16-run-cadical-1.7.4/     ee  446015  380533  178800  201733  340899    43891     484    168   349       0 14761448.5 36784578.1
2024-09-16-run-cadical-2.0.0/     ee  446015  380441  178771  201670  343055    44023     446    168   253       0 14814215.2 36766938.8
```
```
                                            status  total  solved    sat  unsat   best  timeout  memout  error  uniq  disagr  time_cpu    memory
benchmark     config
ABV           2024-09-16-run-cadical-1.7.4/     ok   4982     114     24     90    106        2       0      0     0       0     743.5   12894.5
              2024-09-16-run-cadical-2.0.0/     ok   4982     114     24     90    113        2       0      0     0       0     613.5   11624.4
ABVFP         2024-09-16-run-cadical-1.7.4/     ok     60      13     13      0     10        7       0      0     0       0    2109.0    4719.1
              2024-09-16-run-cadical-2.0.0/     ok     60      13     13      0     12        7       0      0     0       0    2108.8    4169.1
ABVFPLRA      2024-09-16-run-cadical-1.7.4/     ok     77      20     18      2     19       11       0      0     0       0    3314.0    1402.4
              2024-09-16-run-cadical-2.0.0/     ok     77      20     18      2     18       11       0      0     0       0    3314.9    1397.2
ALIA          2024-09-16-run-cadical-1.7.4/     ok   3137     112     15     97    101        4       0      0     0       0    1367.0    8923.9
              2024-09-16-run-cadical-2.0.0/     ok   3137     105     15     90    101        3       0      0     0       0    1013.0    8655.7
ANIA          2024-09-16-run-cadical-1.7.4/     ok     78       2      0      2      2        5       0      0     0       0    1505.9    1058.2
              2024-09-16-run-cadical-2.0.0/     ok     78       2      0      2      2        5       0      0     0       0    1506.0    1056.7
AUFBV         2024-09-16-run-cadical-1.7.4/     ee   1547     205      0    205    142      149      28    158     5       0   70527.4  699616.0
              2024-09-16-run-cadical-2.0.0/     ee   1547     188      0    188    102      183      25    158     2       0   76003.8  688746.2
AUFBVDTLIA    2024-09-16-run-cadical-1.7.4/     ok   1684     393      4    389    384      690       3      0     0       0  207761.1  532500.3
              2024-09-16-run-cadical-2.0.0/     ok   1684     393      4    389    387      690       3      0     0       0  207767.5  525561.1
AUFBVDTNIA    2024-09-16-run-cadical-1.7.4/     ok      8       3      0      3      3        1       0      0     0       0     324.1     123.7
              2024-09-16-run-cadical-2.0.0/     ok      8       3      0      3      3        1       0      0     0       0     325.1     123.0
AUFBVDTNIRA   2024-09-16-run-cadical-1.7.4/     ok   2153     285      0    285    179     1859       0      0     1       0  560424.2 1578200.8
              2024-09-16-run-cadical-2.0.0/     ok   2153     285      0    285    208     1859       0      0     0       0  560419.8 1576122.8
AUFBVFP       2024-09-16-run-cadical-1.7.4/     ee     58      14      0     14     12       11       0      4     0       0    4074.6   32490.9
              2024-09-16-run-cadical-2.0.0/     ee     58      14      0     14      5       11       0      4     0       0    4164.0   34103.1
AUFDTLIA      2024-09-16-run-cadical-1.7.4/     ok    795     697      3    694    683       35       0      0     0       0   10567.8   18602.2
              2024-09-16-run-cadical-2.0.0/     ok    795     697      3    694    684       35       0      0     0       0   10567.1   18540.0
AUFDTLIRA     2024-09-16-run-cadical-1.7.4/     ok  11059    9916      0   9916   9314       62       0      0     0       0   20061.3   67749.3
              2024-09-16-run-cadical-2.0.0/     ok  11059    9914      0   9914   9359       64       0      0     0       0   20621.4   69307.2
AUFDTNIRA     2024-09-16-run-cadical-1.7.4/     ok   1580    1277      0   1277   1212      275       0      0     0       0   82959.7   29422.8
              2024-09-16-run-cadical-2.0.0/     ok   1580    1277      0   1277   1202      274       0      0     0       0   82719.8   29684.8
AUFFPDTNIRA   2024-09-16-run-cadical-1.7.4/     ok    167     138      0    138    130       10       0      0     0       0    3018.6    2660.5
              2024-09-16-run-cadical-2.0.0/     ok    167     138      0    138    134       10       0      0     0       0    3018.5    2667.8
AUFLIA        2024-09-16-run-cadical-1.7.4/     ok   3332    2044     28   2016   1934      418       5      0     0       0  129680.0  425969.2
              2024-09-16-run-cadical-2.0.0/     ok   3332    2044     28   2016   1954      417       6      0     0       0  129636.9  428278.4
AUFLIRA       2024-09-16-run-cadical-1.7.4/     ok  20017   19749      0  19749  19428      112       0      0     0       0   36128.8  112683.6
              2024-09-16-run-cadical-2.0.0/     ok  20017   19749      0  19749  19548      112       0      0     0       0   36122.3  112512.2
AUFNIA        2024-09-16-run-cadical-1.7.4/     ok      3       0      0      0      0        0       0      0     0       0       0.0       5.2
              2024-09-16-run-cadical-2.0.0/     ok      3       0      0      0      0        0       0      0     0       0       0.0       5.2
AUFNIRA       2024-09-16-run-cadical-1.7.4/     ok   1489    1041      0   1041   1009      399       0      0     0       0  120792.0  402801.8
              2024-09-16-run-cadical-2.0.0/     ok   1489    1039      0   1039   1014      401       0      0     0       0  121288.5  387224.1
BV            2024-09-16-run-cadical-1.7.4/     ok   6255    5709    503   5206   5322      546       0      0     4       0  170555.3  292514.5
              2024-09-16-run-cadical-2.0.0/     ok   6255    5716    507   5209   5285      539       0      0     6       0  170552.8  294596.7
BVFP          2024-09-16-run-cadical-1.7.4/     ok    208     108    106      2    100       29       3      0     0       0    9423.9   51624.5
              2024-09-16-run-cadical-2.0.0/     ok    208     108    106      2     96       29       3      0     0       0    9373.5   51461.9
BVFPLRA       2024-09-16-run-cadical-1.7.4/     ok    266     135    111     24    102       99       0      0     0       0   29912.0   20562.3
              2024-09-16-run-cadical-2.0.0/     ok    266     135    111     24    116       99       0      0     0       0   29890.5   20694.2
FP            2024-09-16-run-cadical-1.7.4/     ok   2738    2236    166   2070   1725      452      50      0     1       0  167117.1 1127439.9
              2024-09-16-run-cadical-2.0.0/     ok   2738    2251    168   2083   1628      457      30      0     0       0  170610.1 1155467.3
FPLRA         2024-09-16-run-cadical-1.7.4/     ok     41      23     23      0     18       17       0      0     0       0    5289.3    3579.1
              2024-09-16-run-cadical-2.0.0/     ok     41      23     23      0     20       17       0      0     0       0    5252.3    3527.5
LIA           2024-09-16-run-cadical-1.7.4/     ok    416     416    150    266    395        0       0      0     0       0      85.8    1144.0
              2024-09-16-run-cadical-2.0.0/     ok    416     416    150    266    392        0       0      0     0       0      86.8    1147.2
LRA           2024-09-16-run-cadical-1.7.4/     ok   2457    1911    596   1315   1804      546       0      0     1       0  170491.2   65425.8
              2024-09-16-run-cadical-2.0.0/     ok   2457    1906    594   1312   1821      551       0      0     1       0  170956.7   65983.5
NIA           2024-09-16-run-cadical-1.7.4/     ok    269     210     64    146    193       59       0      0     1       0   18172.3   13565.8
              2024-09-16-run-cadical-2.0.0/     ok    269     207     63    144    190       62       0      0     0       0   18724.0   13771.5
NRA           2024-09-16-run-cadical-1.7.4/     ok   3821    3802      3   3799   3797       19       0      0     0       0    5749.7   12403.8
              2024-09-16-run-cadical-2.0.0/     ok   3821    3802      3   3799   3798       19       0      0     0       0    5749.7   12339.9
QF_ABV        2024-09-16-run-cadical-1.7.4/     ok  15210   14680  10148   4532  13210      473      57      0    65       0  211216.3 1366795.5
              2024-09-16-run-cadical-2.0.0/     ok  15210   14629  10111   4518  12403      538      43      0    10       0  231510.9 1320532.1
QF_ABVFP      2024-09-16-run-cadical-1.7.4/     ok  18141   18110  14174   3936  17117       30       1      0     8       0   27452.1  342081.6
              2024-09-16-run-cadical-2.0.0/     ok  18141   18110  14178   3932  16901       30       1      0     6       0   28093.3  344158.9
QF_ABVFPLRA   2024-09-16-run-cadical-1.7.4/     ok     75      74     69      5     68        1       0      0     0       0     812.5    7408.5
              2024-09-16-run-cadical-2.0.0/     ok     75      75     69      6     63        0       0      0     0       0     765.9    7206.7
QF_ALIA       2024-09-16-run-cadical-1.7.4/     ok    182     145     74     71    114       37       0      0     0       0   15025.6   60154.3
              2024-09-16-run-cadical-2.0.0/     ok    182     151     79     72    106       31       0      0     2       0   14550.0   60650.0
QF_ANIA       2024-09-16-run-cadical-1.7.4/     ok    161     122    109     13    100       39       0      0     0       0   13945.5   20391.4
              2024-09-16-run-cadical-2.0.0/     ok    161     122    109     13     99       39       0      0     0       0   13649.1   20825.0
QF_AUFBV      2024-09-16-run-cadical-1.7.4/     ok     76      42     11     31     38       22      12      0     0       0    8474.2  135953.6
              2024-09-16-run-cadical-2.0.0/     ok     76      44     13     31     37       20      12      0     0       0    7644.6  134064.5
QF_AUFBVFP    2024-09-16-run-cadical-1.7.4/     ok      1       1      1      0      1        0       0      0     0       0       2.4     152.5
              2024-09-16-run-cadical-2.0.0/     ok      1       1      1      0      0        0       0      0     0       0       6.8     162.8
QF_AUFLIA     2024-09-16-run-cadical-1.7.4/     ok   1303    1303    543    760   1192        0       0      0     0       0    1133.0   13265.9
              2024-09-16-run-cadical-2.0.0/     ok   1303    1303    543    760   1132        0       0      0     0       0    1160.3   13277.1
QF_AUFNIA     2024-09-16-run-cadical-1.7.4/     ok     17       9      4      5      6        8       0      0     0       0    2737.7    6754.8
              2024-09-16-run-cadical-2.0.0/     ok     17      10      4      6      7        7       0      0     1       0    2395.6    6672.8
QF_AX         2024-09-16-run-cadical-1.7.4/     ok    551     551    272    279    503        0       0      0     0       0     249.5    2629.9
              2024-09-16-run-cadical-2.0.0/     ok    551     551    272    279    512        0       0      0     0       0     236.7    2620.2
QF_BV         2024-09-16-run-cadical-1.7.4/     ok  46565   43476  17285  26191  36784     2986     103      0    91       0 1278931.9 5648852.8
              2024-09-16-run-cadical-2.0.0/     ok  46565   43444  17279  26165  35053     3021     100      0    31       0 1290196.8 5647102.3
QF_BVFP       2024-09-16-run-cadical-1.7.4/     ok  17251   17243  14063   3180  16134        8       0      0     1       0    8404.2  242738.2
              2024-09-16-run-cadical-2.0.0/     ok  17251   17244  14065   3179  16020        7       0      0     0       0    8647.2  242811.5
QF_BVFPLRA    2024-09-16-run-cadical-1.7.4/     ok    182     163    123     40    143       19       0      0     0       0    6888.9   12172.5
              2024-09-16-run-cadical-2.0.0/     ok    182     163    123     40    127       19       0      0     0       0    6385.0   12415.3
QF_DT         2024-09-16-run-cadical-1.7.4/     ok   8707    8274   3676   4598   8192      433       0      0     6       0  138499.1  286635.4
              2024-09-16-run-cadical-2.0.0/     ok   8707    8268   3671   4597   8163      439       0      0     4       0  141263.8  281569.9
QF_FP         2024-09-16-run-cadical-1.7.4/     ok  40417   40342  20243  20099  40246       68       7      0     4       0   27933.2  238674.1
              2024-09-16-run-cadical-2.0.0/     ok  40417   40333  20237  20096  40258       77       7      0     2       0   30523.3  243711.2
QF_FPLRA      2024-09-16-run-cadical-1.7.4/     ok     60      50     47      3     48       10       0      0     1       0    3486.8   20695.9
              2024-09-16-run-cadical-2.0.0/     ok     60      49     46      3     44       11       0      0     0       0    3824.3   20649.4
QF_IDL        2024-09-16-run-cadical-1.7.4/     ok   2616    1647    948    699   1079      959      10      0     8       0  344127.2  886225.2
              2024-09-16-run-cadical-2.0.0/     ok   2616    1660    950    710   1112      946      10      0     5       0  341320.6  886987.8
QF_LIA        2024-09-16-run-cadical-1.7.4/     ok  13495   10077   6942   3135   8341     3347      71      0    22       0 1151528.6 4099969.7
              2024-09-16-run-cadical-2.0.0/     ok  13495   10077   6942   3135   8307     3349      69      0    19       0 1149522.4 4158092.8
QF_LIRA       2024-09-16-run-cadical-1.7.4/     ok      7       5      1      4      2        2       0      0     0       0     633.7    1027.2
              2024-09-16-run-cadical-2.0.0/     ok      7       5      1      4      4        2       0      0     0       0     638.2     966.0
QF_LRA        2024-09-16-run-cadical-1.7.4/     ok   1794    1604    973    631   1254      190       0      0     5       0   94481.1  229335.6
              2024-09-16-run-cadical-2.0.0/     ok   1794    1602    963    639   1212      192       0      0     1       0   95235.4  227818.3
QF_NIA        2024-09-16-run-cadical-1.7.4/     ok  27657   15114  10551   4563   9944    12538       5      0    93       0 4095398.7 3991727.1
              2024-09-16-run-cadical-2.0.0/     ok  27657   15143  10594   4549  10011    12510       4      0   121       0 4103322.6 3949637.3
QF_NIRA       2024-09-16-run-cadical-1.7.4/     ok      3       2      0      2      2        1       0      0     0       0     418.5    1087.0
              2024-09-16-run-cadical-2.0.0/     ok      3       2      0      2      1        1       0      0     0       0     472.9    1094.1
QF_NRA        2024-09-16-run-cadical-1.7.4/     ok  12213   11282   5513   5769  10408      929       2      0     4       0  325200.4  418954.8
              2024-09-16-run-cadical-2.0.0/     ok  12213   11299   5529   5770  10420      912       2      0     6       0  321625.6  419870.2
QF_RDL        2024-09-16-run-cadical-1.7.4/     ok    258     202    100    102    132       56       0      0     1       0   22757.9   20734.9
              2024-09-16-run-cadical-2.0.0/     ok    258     202     99    103    135       56       0      0     0       0   22259.9   20536.2
QF_S          2024-09-16-run-cadical-1.7.4/     ok  18994   18840  12070   6770  18260      146       8      0     0       0   50705.7  170146.7
              2024-09-16-run-cadical-2.0.0/     ok  18994   18841  12070   6771  18345      145       8      0     1       0   50651.2  167375.1
QF_SLIA       2024-09-16-run-cadical-1.7.4/     ok  87355   83920  51528  32392  72211     3423       9      1     2       0 1198990.2 3957852.9
              2024-09-16-run-cadical-2.0.0/     ok  87355   83891  51500  32391  77457     3452       9      1     6       0 1203556.8 3964664.4
QF_SNIA       2024-09-16-run-cadical-1.7.4/     ok     70      70     70      0     70        0       0      0     0       0       0.0     108.4
              2024-09-16-run-cadical-2.0.0/     ok     70      70     70      0     70        0       0      0     0       0       0.0     110.0
QF_UF         2024-09-16-run-cadical-1.7.4/     ok   7505    7495   3142   4353   6440       10       0      0     0       0    7308.1   53177.5
              2024-09-16-run-cadical-2.0.0/     ok   7505    7494   3142   4352   6263       11       0      0     1       0    7988.2   53363.3
QF_UFBV       2024-09-16-run-cadical-1.7.4/     ok   1523    1324    715    609   1012      133      66      0     6       0   62135.7  957942.8
              2024-09-16-run-cadical-2.0.0/     ok   1523    1317    712    605    919      139      67      0     5       0   63900.8  965241.5
QF_UFBVDT     2024-09-16-run-cadical-1.7.4/     ok     79      36     31      5     21       15      28      0     2       0    8579.2  338291.3
              2024-09-16-run-cadical-2.0.0/     ok     79      34     29      5     15       17      28      0     0       0    9080.6  335648.1
QF_UFDT       2024-09-16-run-cadical-1.7.4/     ok    208      83     83      0     45      125       0      0     6       0   43300.3   63527.6
              2024-09-16-run-cadical-2.0.0/     ok    208      87     86      1     50      121       0      0     8       0   41884.4   63478.3
QF_UFDTLIA    2024-09-16-run-cadical-1.7.4/     ok     79      37     30      7     20       37       5      0     0       0   13344.5   85785.3
              2024-09-16-run-cadical-2.0.0/     ok     79      37     30      7     21       35       7      0     0       0   13300.1   87716.8
QF_UFDTLIRA   2024-09-16-run-cadical-1.7.4/     ok    146     146     80     66    146        0       0      0     0       0       0.0     199.1
              2024-09-16-run-cadical-2.0.0/     ok    146     146     80     66    146        0       0      0     0       0       0.0     199.7
QF_UFDTNIA    2024-09-16-run-cadical-1.7.4/     ok     83      27     18      9     16       56       0      0     0       0   17703.1   20061.0
              2024-09-16-run-cadical-2.0.0/     ok     83      29     20      9     18       54       0      0     0       0   17354.6   19831.2
QF_UFFP       2024-09-16-run-cadical-1.7.4/     ok      2       2      0      2      2        0       0      0     0       0       0.7      72.5
              2024-09-16-run-cadical-2.0.0/     ok      2       2      0      2      2        0       0      0     0       0       0.7      72.5
QF_UFFPDTNIRA 2024-09-16-run-cadical-1.7.4/     ok    154     154     24    130    142        0       0      0     0       0      82.6    5841.5
              2024-09-16-run-cadical-2.0.0/     ok    154     154     24    130    127        0       0      0     0       0     100.8    5841.2
QF_UFIDL      2024-09-16-run-cadical-1.7.4/     ok    638     481    206    275    365      157       0      0     0       0   53774.7   91361.6
              2024-09-16-run-cadical-2.0.0/     ok    638     480    206    274    279      158       0      0     1       0   54529.3   91346.9
QF_UFLIA      2024-09-16-run-cadical-1.7.4/     ok    662     621    433    188    469       35       6      0     1       0   15270.8  105634.1
              2024-09-16-run-cadical-2.0.0/     ok    662     620    432    188    541       35       7      0     0       0   14450.5  110685.1
QF_UFLRA      2024-09-16-run-cadical-1.7.4/     ok   1285    1273    762    511   1150        9       3      0     0       0    4322.0  100988.1
              2024-09-16-run-cadical-2.0.0/     ok   1285    1274    763    511   1122        8       3      0     1       0    4244.1  101324.4
QF_UFNIA      2024-09-16-run-cadical-1.7.4/     ok    818     565    353    212    501      252       1      0     4       0   80565.9  230735.0
              2024-09-16-run-cadical-2.0.0/     ok    818     562    347    215    494      255       1      0     5       0   81827.2  229085.9
QF_UFNRA      2024-09-16-run-cadical-1.7.4/     ok     65      44     27     17     25       21       0      0     1       0    8827.9   34948.5
              2024-09-16-run-cadical-2.0.0/     ok     65      40     23     17     31       25       0      0     1       0    9250.5   34379.2
UF            2024-09-16-run-cadical-1.7.4/     ok   7635    3196      1   3195   2741     3716       0      0     1       0 1123217.8 1262019.7
              2024-09-16-run-cadical-2.0.0/     ok   7635    3196      1   3195   2726     3716       0      0     1       0 1123624.6 1268793.7
UFBV          2024-09-16-run-cadical-1.7.4/     ok    193      60      8     52     58        3       1      0     0       0    1610.5   27674.4
              2024-09-16-run-cadical-2.0.0/     ok    193      60      8     52     56        3       1      0     0       0    1619.1   27551.6
UFBVDT        2024-09-16-run-cadical-1.7.4/     ok      1       0      0      0      0        0       0      0     0       0      16.3     566.4
              2024-09-16-run-cadical-2.0.0/     ok      1       0      0      0      0        0       0      0     0       0       7.5     555.8
UFBVFP        2024-09-16-run-cadical-1.7.4/     ok      2       1      0      1      1        0       0      0     0       0      13.2     818.4
              2024-09-16-run-cadical-2.0.0/     ok      2       1      0      1      1        0       0      0     0       0      12.4     708.9
UFBVLIA       2024-09-16-run-cadical-1.7.4/     ok    208       0      0      0      0        0       0      0     0       0     187.7   12754.3
              2024-09-16-run-cadical-2.0.0/     ok    208       0      0      0      0        0       0      0     0       0     190.9   12755.8
UFDT          2024-09-16-run-cadical-1.7.4/     ee   4610    2198      0   2198   1883     1983       0      3     0       0  601865.8  974645.3
              2024-09-16-run-cadical-2.0.0/     ee   4610    2195      0   2195   1908     1986       0      3     1       0  602470.2  980492.6
UFDTLIA       2024-09-16-run-cadical-1.7.4/     ok    333      76      0     76     70      245       0      0     0       0   73743.8  374859.2
              2024-09-16-run-cadical-2.0.0/     ok    333      76      0     76     71      245       0      0     0       0   73738.2  376412.4
UFDTLIRA      2024-09-16-run-cadical-1.7.4/     ok   7752    5938    100   5838   5759        6       0      0     0       0    2594.5   30264.4
              2024-09-16-run-cadical-2.0.0/     ok   7752    5938    100   5838   5773        6       0      0     0       0    2575.9   30156.4
UFDTNIA       2024-09-16-run-cadical-1.7.4/     ok      2       1      0      1      1        0       0      0     0       0      11.7      90.2
              2024-09-16-run-cadical-2.0.0/     ok      2       1      0      1      1        0       0      0     0       0      11.5      90.4
UFDTNIRA      2024-09-16-run-cadical-1.7.4/     ok   4452    4096      0   4096   3794      287       0      0     0       0   87735.5   63290.0
              2024-09-16-run-cadical-2.0.0/     ok   4452    4096      0   4096   3771      285       0      0     0       0   87043.0   63092.8
UFFPDTNIRA    2024-09-16-run-cadical-1.7.4/     ok    796     671     39    632    602       20       0      0     0       0    7533.2   29939.1
              2024-09-16-run-cadical-2.0.0/     ok    796     671     39    632    569       20       0      0     0       0    7359.6   28990.4
UFIDL         2024-09-16-run-cadical-1.7.4/     ok     68      56      0     56     55        5       0      0     0       0    1503.9    2579.5
              2024-09-16-run-cadical-2.0.0/     ok     68      56      0     56     53        5       0      0     0       0    1503.9    2573.3
UFLIA         2024-09-16-run-cadical-1.7.4/     ok  10196    7496      0   7496   6392     1940       0      1     1       0  599256.4  939816.4
              2024-09-16-run-cadical-2.0.0/     ok  10196    7495      0   7495   6425     1941       0      1     0       0  598789.2  934304.7
UFLRA         2024-09-16-run-cadical-1.7.4/     ok     15      10      0     10     10        0       0      0     0       0       0.0      21.8
              2024-09-16-run-cadical-2.0.0/     ok     15      10      0     10     10        0       0      0     0       0       0.0      22.3
UFNIA         2024-09-16-run-cadical-1.7.4/     ok  14181    6445   1383   5062   5384     3244       0      1     3       0 1070089.0 3233898.2
              2024-09-16-run-cadical-2.0.0/     ok  14181    6442   1381   5061   5400     3246       0      1     5       0 1070230.6 3227597.3
UFNIRA        2024-09-16-run-cadical-1.7.4/     ok    266      80      0     80     72       50       0      0     0       0   15263.2   40164.4
              2024-09-16-run-cadical-2.0.0/     ok    266      80      0     80     72       50       0      0     0       0   15260.3   40275.7
```
@arminbiere @kfazekas @m-fleury
---
 cmake/FindCaDiCaL.cmake                                   | 8 +++++---
 src/prop/cadical.cpp                                      | 3 +++
 .../cli/regress0/proofs/proj-issue528-ufpfchecker.smt2    | 3 +--
 .../cli/regress0/proofs/proj-issue529-open-pf.smt2        | 3 +--
 test/regress/cli/regress1/nl/proj-issue530.smt2           | 1 -
 test/regress/cli/run_regression.py                        | 5 ++++-
 6 files changed, 14 insertions(+), 9 deletions(-)

diff --git a/cmake/FindCaDiCaL.cmake b/cmake/FindCaDiCaL.cmake
index 9ef3ef717..61745f2bf 100644
--- a/cmake/FindCaDiCaL.cmake
+++ b/cmake/FindCaDiCaL.cmake
@@ -85,8 +85,8 @@ if(NOT CaDiCaL_FOUND_SYSTEM)
   include(CheckSymbolExists)
   include(ExternalProject)
 
-  set(CaDiCaL_VERSION "rel-1.7.4")
-  set(CaDiCaL_CHECKSUM "866c8a1332ff1ad5dc7ad403bdef3164420f3f947816b5c9509aad1d18ada7a1")
+  set(CaDiCaL_VERSION "rel-2.0.0")
+  set(CaDiCaL_CHECKSUM "9afe5f6439442d854e56fc1fac3244ce241dbb490735939def8fd03584f89331")
 
   # avoid configure script and instantiate the makefile manually the configure
   # scripts unnecessarily fails for cross compilation thus we do the bare
@@ -121,6 +121,7 @@ if(NOT CaDiCaL_FOUND_SYSTEM)
     set(USE_EMAR  "-e s,ar rc,emar rc,")
   endif()
 
+  set(CaDiCaL_SOURCE_DIR <SOURCE_DIR>)
   ExternalProject_Add(
     CaDiCaL-EP
     ${COMMON_EP_CONFIG}
@@ -133,7 +134,8 @@ if(NOT CaDiCaL_FOUND_SYSTEM)
             <SOURCE_DIR>/build/makefile
     COMMAND
       sed -i.orig -e "s,@CXX@,${CMAKE_CXX_COMPILER}," -e
-      "s,@CXXFLAGS@,${CaDiCaL_CXXFLAGS}," -e "s,@MAKEFLAGS@,," ${USE_EMAR}
+      "s,@CXXFLAGS@,${CaDiCaL_CXXFLAGS}," -e
+      "s,@ROOT@,${CaDiCaL_SOURCE_DIR}," -e "s,@CONTRIB@,no," ${USE_EMAR}
       <SOURCE_DIR>/build/makefile
     BUILD_COMMAND ${make_cmd} -C <SOURCE_DIR>/build libcadical.a
     INSTALL_COMMAND ${CMAKE_COMMAND} -E copy <SOURCE_DIR>/build/libcadical.a
diff --git a/src/prop/cadical.cpp b/src/prop/cadical.cpp
index bc07bdf6b..2eb33662d 100644
--- a/src/prop/cadical.cpp
+++ b/src/prop/cadical.cpp
@@ -1012,6 +1012,9 @@ void CadicalSolver::init()
   {
     d_solver->set("walk", 0);
     d_solver->set("lucky", 0);
+    // ilb currently does not play well with user propagators
+    d_solver->set("ilb", 0);
+    d_solver->set("ilbassumptions", 0);
   }
 
   d_solver->set("quiet", 1);  // CaDiCaL is verbose by default
diff --git a/test/regress/cli/regress0/proofs/proj-issue528-ufpfchecker.smt2 b/test/regress/cli/regress0/proofs/proj-issue528-ufpfchecker.smt2
index 311d220ba..570991df6 100644
--- a/test/regress/cli/regress0/proofs/proj-issue528-ufpfchecker.smt2
+++ b/test/regress/cli/regress0/proofs/proj-issue528-ufpfchecker.smt2
@@ -1,6 +1,5 @@
 (set-logic QF_ALL)
 (set-info :status unsat)
-(set-option :check-proofs true)
 (set-option :proof-check eager)
 (assert (> real.pi (seq.nth (seq.++ (seq.++ (seq.unit real.pi) (seq.rev (seq.++ (seq.unit real.pi) (seq.unit real.pi)))) (seq.unit real.pi) (seq.unit real.pi)) (seq.len (seq.++ (seq.unit real.pi) (seq.unit real.pi))))))
-(check-sat)
\ No newline at end of file
+(check-sat)
diff --git a/test/regress/cli/regress0/proofs/proj-issue529-open-pf.smt2 b/test/regress/cli/regress0/proofs/proj-issue529-open-pf.smt2
index 19d57007d..1c8468490 100644
--- a/test/regress/cli/regress0/proofs/proj-issue529-open-pf.smt2
+++ b/test/regress/cli/regress0/proofs/proj-issue529-open-pf.smt2
@@ -2,7 +2,6 @@
 ; EXPECT: unsat
 ; EXPECT: unsat
 (set-logic ALL)
-(set-option :check-proofs true)
 (set-option :proof-check eager)
 (set-option :incremental true)
 (declare-const x (Set Bool))
@@ -10,4 +9,4 @@
 (assert (set.subset x (set.choose (set.singleton (set.minus (set.singleton (set.choose x)) (set.inter x (set.singleton (set.choose x))))))))
 (check-sat)
 (check-sat-assuming ((set.choose (set.singleton (set.choose x)))))
-(check-sat)
\ No newline at end of file
+(check-sat)
diff --git a/test/regress/cli/regress1/nl/proj-issue530.smt2 b/test/regress/cli/regress1/nl/proj-issue530.smt2
index 8dbc4f747..07c6e8100 100644
--- a/test/regress/cli/regress1/nl/proj-issue530.smt2
+++ b/test/regress/cli/regress1/nl/proj-issue530.smt2
@@ -1,6 +1,5 @@
 (set-logic ALL)
 (set-info :status unsat)
-(set-option :check-proofs true)
 (set-option :proof-check eager)
 (declare-const x Real)
 (assert (= real.pi (/ real.pi (tan (to_real (to_int x))))))
diff --git a/test/regress/cli/run_regression.py b/test/regress/cli/run_regression.py
index ca15790d2..bf386f170 100755
--- a/test/regress/cli/run_regression.py
+++ b/test/regress/cli/run_regression.py
@@ -181,7 +181,10 @@ class ProofTester(Tester):
         return super().run_internal(
             benchmark_info._replace(
                 command_line_args=benchmark_info.command_line_args +
-                ["--check-proofs", "--proof-granularity=theory-rewrite", "--proof-check=lazy"]
+                ["--sat-solver=minisat",
+                 "--check-proofs",
+                 "--proof-granularity=theory-rewrite",
+                 "--proof-check=lazy"]
             )
         )
 
-- 
2.47.1

